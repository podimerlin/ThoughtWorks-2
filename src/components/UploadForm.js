import React, { Component } from 'react'

class UploadForm extends Component {
    constructor() {
        super();

        this.state = { file : null};
        this.handleSubmit = this.handleSubmit.bind(this);
        this.handleChange = this.handleChange.bind(this);
        this.inputData = null;
    }

    handleChange(e) {
        this.setState({file: e.target.files[0]});
    } 

    handleSubmit(e) {
        e.preventDefault();

        var inputData = '';
        var file = this.state.file;
        if (file) {
            var reader = new FileReader();
            reader.onload = (function(file) {
                return function(e) {
                    inputData = e.target.result;
                    this.props.onUpload(inputData);
                };
            })(file).bind(this);
            reader.readAsText(file);            
        }

    }

    render() {
        return(
            <form onSubmit={this.handleSubmit}>
                <label> Enter File: <br/>
                    <input type="file" id="fileinput" onChange={this.handleChange}/>    
                </label>
                <input type="submit" value="Submit" className="btn-upload" />
            </form>
        );
    }
}

export default UploadForm;