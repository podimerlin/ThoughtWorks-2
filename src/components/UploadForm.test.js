import React from 'react';
import { shallow } from 'enzyme';
import UploadForm from './UploadForm';



describe('UploadForm', () => {    
    const mockOnUpload = jest.fn();
    const file = new File([""], "filename");
    const form = shallow(<UploadForm onUpload={mockOnUpload}/>);
    const event = {
              preventDefault: jest.fn(),
              target: {files : [file]}
            };

    const input = 'Writing Fast Tests Against Enterprise Rails 60min';

    it('renders correctly', () => {
        expect(form).toMatchSnapshot();
    });

    it('initialize the `state` with a null file', () => {
        expect(form.state().file).toEqual(null);
    });

    describe('when clicking `upload-file` button', () => {
        beforeEach(() => {
          form.find('#fileinput').simulate('change', event);
          form.find('form').simulate('submit', event);
        });

        it('calls the `state` file to be not null CallBack', () => {
          expect(form.state().file).not.toBeNull();
        });

    });
})
