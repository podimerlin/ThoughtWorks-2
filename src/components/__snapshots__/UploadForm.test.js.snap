// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UploadForm renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UploadForm
    onUpload={[Function]}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label>
           Enter File: 
          <br />
          <input
                    id="fileinput"
                    onChange={[Function]}
                    type="file"
          />
</label>,
        <input
          className="btn-upload"
          type="submit"
          value="Submit"
/>,
      ],
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            " Enter File: ",
            <br />,
            <input
              id="fileinput"
              onChange={[Function]}
              type="file"
/>,
          ],
        },
        "ref": null,
        "rendered": Array [
          " Enter File: ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "br",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "id": "fileinput",
              "onChange": [Function],
              "type": "file",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "btn-upload",
          "type": "submit",
          "value": "Submit",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
             Enter File: 
            <br />
            <input
                        id="fileinput"
                        onChange={[Function]}
                        type="file"
            />
</label>,
          <input
            className="btn-upload"
            type="submit"
            value="Submit"
/>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              " Enter File: ",
              <br />,
              <input
                id="fileinput"
                onChange={[Function]}
                type="file"
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            " Enter File: ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "id": "fileinput",
                "onChange": [Function],
                "type": "file",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "btn-upload",
            "type": "submit",
            "value": "Submit",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
